azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
+        Executors.newFixedThreadPool(activeExecutors.size());
+      for (final Executor executor : activeExecutors) {
+
+        // execute each executorInfo refresh task
+        taskExecutors.execute(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              String jsonResponse =
+                callExecutorForJsonString(executor.getHost(),
+                  executor.getPort(), "/serverstastics", null);
+              executor.setExecutorInfo(ExecutorInfo
+                .fromJSONString(jsonResponse));
+            } catch (Exception e) {
+              logger.error("Failed to update ExecutorInfo executorId :"
+                + executor, e);
+            }
+          }
+        });
+      }
+
+      taskExecutors.shutdown();
+      try {
+        // wait 5 seconds for all executors to be refreshed
+        taskExecutors.awaitTermination(5, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
+        logger.error("Timed out while waiting for executorInfo refresh", e);
```

Comment:
InterruptedException is raised when the current thread (the thread hosts the Executor) is terminated, it does not indicate that the threads in pool incurred timeout while awaited for finishing.  the function call  awaitTermination() actually returns a Boolean value which indicates if the wait time is expired. You may want to read that value and write log accordingly. also it might be nicer to tell which Executor timed out as it might be a key piece of data for debugging.